김현수/mvvmstudy/app/src/main/java/com/example/mvvmstudy/UI/MainActivity.kt

```diff
@@ -0,0 +1,176 @@
+package com.example.mvvmstudy.UI
+
+import android.annotation.SuppressLint
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import retrofit2.Callback
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.SearchView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.mvvmstudy.R
+import com.example.mvvmstudy.IO.RetrofitClass
+import com.example.mvvmstudy.data.Repositories
+import com.example.mvvmstudy.databinding.ActivityMainBinding
+import org.jetbrains.anko.indeterminateProgressDialog
+import retrofit2.Call
+import retrofit2.Response
+
+class MainActivity : AppCompatActivity() {
+
+    private var page = 0
+    private var search =""
+
+    private var _binding : ActivityMainBinding?= null
+    private val binding get() = _binding!!
+    private val adapter = RepositoryAdapter(this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        binding.listRepo.adapter = adapter
+        binding.listRepo.layoutManager = LinearLayoutManager(this)
+
+        initScrollListener()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.main_menu,menu)
+        val searchItem = menu?.findItem(R.id.item_search)
+        val searchView = searchItem?.actionView as SearchView
+        searchView.queryHint = "Search.."
+
+        searchView.setOnQueryTextListener(object:SearchView.OnQueryTextListener{
+            override fun onQueryTextSubmit(p: String?): Boolean {
+                if(search != p.toString()) page = 0
+               fetching(p.toString())
+                search = p.toString()
+
+                // 요녀석 땜에 꽤 고생
+                searchView.clearFocus()
+               return false
+            }
+            override fun onQueryTextChange(p0: String?): Boolean {
+                return false
+            }
+        })
+
+        return super.onCreateOptionsMenu(menu)
+    }
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when(item?.itemId){
+            android.R.id.home ->{
+                finish()
+                return true
+            }
+            else ->{
+                return super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+
+    fun fetching(q: String){
+        RetrofitClass.retrofitService
+            .searchRepositories(q,getPage(),50)
```

Comment:
요기도 code format되면 좋을 것 같아요!
```suggestion
            .searchRepositories(q,getPage(), 50)
```